### PR TITLE
Refs [TASK][34577] Show list ingredient and nutrient

### PR DESCRIPTION
**Pull request by phong10101998**
## Related Tickets
- [#Task 34577: Show list ingredient and nutrient](https://edu-redmine.sun-asterisk.vn/issues/34577)
## WHAT
- Show list ingredient and nutrient
## Evidence (Screenshot or Video)
List nutrient
image
List Ingredient
image

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/layout/fragment_bottom_sheet_nutrient.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    android:layout_height="wrap_content"
+    android:paddingBottom="@dimen/dp_5">
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recyclerViewBottomSheetNutrient"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/dp_15"
+        android:layout_marginBottom="@dimen/dp_20"
+        android:dividerHeight="8dp"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/fragment_bottom_sheet_nutrient.xml]: hard code. chắc phải thu $ mới nhớ

File: app/src/main/res/layout/fragment_bottom_sheet_ingredient.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    android:layout_height="wrap_content"
+    android:paddingBottom="@dimen/dp_5">
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recyclerViewBottomSheet"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/dp_20"
+        android:layout_marginBottom="@dimen/dp_20"
+        android:dividerHeight="8dp"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/fragment_bottom_sheet_ingredient.xml]: hard code

File: app/src/main/java/com/sun/spoonacular/ui/detail/adapter/NutrientDiffUtil.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.sun.spoonacular.ui.detail.adapter
+
+import androidx.recyclerview.widget.DiffUtil
+import com.sun.spoonacular.data.model.Nutrient
+
+class NutrientDiffUtil : DiffUtil.ItemCallback<Nutrient>() {
+    override fun areItemsTheSame(oldItem: Nutrient, newItem: Nutrient): Boolean {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/detail/adapter/NutrientDiffUtil.kt]:
```suggestion
class NutrientDiffUtil : DiffUtil.ItemCallback<Nutrient>() {

    override fun areItemsTheSame(oldItem: Nutrient, newItem: Nutrient): Boolean {
```

File: app/src/main/java/com/sun/spoonacular/ui/detail/adapter/NutrientAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        return ItemNutrientViewHolder(itemView)
+    }
+
+    override fun onBindViewHolder(holder: ItemNutrientViewHolder, position: Int) {
+        holder.binData(getItem(position))
+    }
+
+    class ItemNutrientViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+
+        @SuppressLint("SetTextI18n")
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/detail/adapter/NutrientAdapter.kt:28]: lỗi chi đây thanh niên?

File: app/src/main/java/com/sun/spoonacular/ui/detail/adapter/IngredientDiffUtil.kt [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+class IngredientDiffUtil : DiffUtil.ItemCallback<Ingredient>() {
+    override fun areItemsTheSame(oldItem: Ingredient, newItem: Ingredient): Boolean {
+        return oldItem.id == newItem.id
+    }
+
+    override fun areContentsTheSame(oldItem: Ingredient, newItem: Ingredient): Boolean {
+        return oldItem == newItem
+    }
+}
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/detail/adapter/IngredientDiffUtil.kt]: enter end line

File: app/src/main/java/com/sun/spoonacular/ui/detail/adapter/IngredientDiffUtil.kt [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package com.sun.spoonacular.ui.detail.adapter
+
+import androidx.recyclerview.widget.DiffUtil
+import com.sun.spoonacular.data.model.Ingredient
+
+class IngredientDiffUtil : DiffUtil.ItemCallback<Ingredient>() {
+    override fun areItemsTheSame(oldItem: Ingredient, newItem: Ingredient): Boolean {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/detail/adapter/IngredientDiffUtil.kt]:
```suggestion
class IngredientDiffUtil : DiffUtil.ItemCallback<Ingredient>() {

    override fun areItemsTheSame(oldItem: Ingredient, newItem: Ingredient): Boolean {
```

File: app/src/main/java/com/sun/spoonacular/ui/detail/DetailRecipeFragment.kt [hunk 7]
```diff
@@ -67,6 +67,17 @@ class DetailRecipeFragment : Fragment() {
         viewPagerStep.adapter = stepAdapter
         adapter.submitList(recipesSimilar)
         recyclerViewSimilarRecipe.adapter = adapter
+        buttonNutrient.setOnClickListener {
+            recipeDetail?.nutrients?.let {
+                NutrientFragment.newInstance(it.nutrients as ArrayList<Nutrient>)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/detail/DetailRecipeFragment.kt]: convert sang dạng list rồi bắn qua e

File: app/src/main/java/com/sun/spoonacular/ui/detail/adapter/NutrientAdapter.kt (hunk 4 above)
Comment (phong10101998) [reply app/src/main/java/com/sun/spoonacular/ui/detail/adapter/NutrientAdapter.kt:28]: > lỗi chi đây thanh niên?
Chỗ textAmount á a, anh nói không cần đưa vô file String á a, nên em bỏ File string rồi thêm cái anotation

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**